zone/client.cpp

```diff
@@ -4269,6 +4269,32 @@ void Client::KeyRingList()
 	}
 }
 
+bool Client::IsLevelFirst(uint32 p_race, uint32 p_class, uint16 level) {
+
+	std::string query = StringFormat("SELECT count(*) FROM server_first_levels WHERE level = '%lu' AND race = '%lu' AND class = '%lu'", level, p_race, p_class);
+	auto results = database.QueryDatabase(query);
+	if (!results.Success()) {
+		return false;
+	}
+
+	auto row = results.begin();
+	if (!atoi(row[0]))
+		return false;
+
+	return true;
+}
+
+void Client::LevelFirst(uint32 p_race, uint32 p_class, uint16 level) {
+
+	std::string query = StringFormat("INSERT INTO server_first_levels "
```

Comment:
You can use fmt like this:
QueryDatabase(fmt::format("INSERT INTO  server_first_levels 
   (char_name, leveled_date, race, class, level, account_status) 
   VALUES ('{}',UNIX_TIMESTAMP(), {}, {}, {}, {})", GetName(), p_race, p_class, level, Admin());

Reply:
> QueryDatabase(fmt::format("INSERT INTO server_first_levels
> (char_name, leveled_date, race, class, level, account_status)
> VALUES ('{}',UNIX_TIMESTAMP(), {}, {}, {}, {})", GetName(), p_race, p_class, level, Admin());

A closing bracket was missing

`QueryDatabase(fmt::format("INSERT INTO server_first_levels (char_name, leveled_date, race, class, level, account_status) VALUES ('{}',UNIX_TIMESTAMP(), {}, {}, {}, {})", GetName(), p_race, p_class, level, Admin()));`